### PR TITLE
Add disable/enable to Corpus

### LINKED ISSUE #2410: Introduce API to disable and enable testcases
Currently there's no way to toggle a testcase's enabled status. Introduce something like:

``corpus.disable/enable()`` which moves it from the enabled to the disabled map.

### PR DESCRIPTION
**Pull request by p13l13d13**
- **Add enable/disable methods to Corpus**

## Description

Added enable/disable methods to the API, so it's now possible to move testcases properly between enabled/disabled maps.
AFAIU, moving to an opposite map and saving (in case of ondisk corpus) is pretty much enough.
Added a pretty basic test to check some basic functionality and corner cases.

Closes #2410. 

## Checklist

- [*] I have run `./scripts/precommit.sh` and addressed all comments

### CODE REVIEW

File: libafl_libfuzzer/runtime/src/corpus.rs [hunk 1]
```diff
@@ -251,6 +251,13 @@ where
             res => res,
         }
     }
+
+    fn disable(&mut self, _: CorpusId) -> Result<(), Error> {
+        unimplemented!("disable is not implemented and not used in this corpus variant!");
```
Comment (domenukk) [inline libafl_libfuzzer/runtime/src/corpus.rs:256]: Instead of a dummy implementation, could we just have a `DisableableCorpus` trait (or a better name, lol), that is not implemented for this specific corpus? That way this will be caught at compile time instead

Comment (p13l13d13) [reply libafl_libfuzzer/runtime/src/corpus.rs]: I've tried my best to pick a better name...

File: libafl/src/corpus/inmemory_ondisk.rs [hunk 2]
```diff
@@ -20,7 +20,7 @@ use libafl_bolts::compress::GzipCompressor;
 use serde::{Deserialize, Serialize};
 
 use super::{
-    HasTestcase,
+    HasCorpusEnablementOperations, HasTestcase,
```
Comment (domenukk) [inline libafl/src/corpus/inmemory_ondisk.rs:23]: `trait EnableDisableCorpus`?

Comment (domenukk) [reply libafl/src/corpus/inmemory_ondisk.rs:23]: or just `trait EnableDisable`

Comment (domenukk) [reply libafl/src/corpus/inmemory_ondisk.rs:23]: `trait Disableable`

Comment (domenukk) [reply libafl/src/corpus/inmemory_ondisk.rs:23]: (it's not a `Has` because it doesn't own anything here)

**Comment by domenukk** [comment]
Thank you!
